core/src/main/java/org/elasticsearch/ingest/core/Processor.java

```diff
@@ -49,7 +54,6 @@
          * Implementations are responsible for removing the used keys, so that after creating a pipeline ingest can
          * verify if all configurations settings have been used.
          */
-        P create(Map<String, Object> config) throws Exception;
-
+        P create(String tag, Map<String, Object> config) throws Exception;
```

Comment:
Can we remove `tag` as parameter here and make it part of the `config` parameter? In the end it is an optional argument and it feels that the this should be part of config, this way the signature of this method remains clean.